src/terraform_gui.cpp

```diff
@@ -118,6 +118,28 @@ bool GUIPlaceProcDragXY(ViewportDragDropSelectionProcess proc, TileIndex start_t
 		case DDSP_DEMOLISH_AREA:
 			DoCommandP(end_tile, start_tile, _ctrl_pressed ? 1 : 0, CMD_CLEAR_AREA | CMD_MSG(STR_ERROR_CAN_T_CLEAR_THIS_AREA), CcPlaySound_EXPLOSION);
 			break;
+		case DDSP_DEMOLISH_TREES:
+			// loop through every tile and send a demolish command for each tree
+			// orthogonal area
+			if (!_ctrl_pressed) {
+				OrthogonalTileArea square_area = OrthogonalTileArea(start_tile, end_tile);
+				TILE_AREA_LOOP(curr_tile, square_area)
+				{
+					if (GetTileType(curr_tile) == MP_TREES) {
+						DoCommandP(curr_tile, curr_tile, 0, CMD_CLEAR_AREA | CMD_MSG(STR_ERROR_CAN_T_CLEAR_THIS_AREA), CcPlaySound_EXPLOSION);
+					}
+				}
+			// diagonal area (not supported)
+			} else {
+				DiagonalTileArea diagonal_area = DiagonalTileArea(start_tile, end_tile);
+				DIAGONAL_TILE_AREA_LOOP(curr_tile, diagonal_area)
+				{
+					if (GetTileType(curr_tile) == MP_TREES) {
+						DoCommandP(curr_tile, curr_tile, 0, CMD_CLEAR_AREA | CMD_MSG(STR_ERROR_CAN_T_CLEAR_THIS_AREA), CcPlaySound_EXPLOSION);
+					}
+				}
+			}
+			break;
```

Comment:
Oh I didn't know you could manipulate DoCommandP like that, thanks :)

Reply:
Wait but I want to keep compatibility with trunk servers - changing/creating commands would adversely affect that right?

Reply:
@Hexus-One Writing a patch for trunc and writing a patch compatible with trunc are two entirely different things. As you're making a trunc PR here you shouldn't be worrying about trunc compatibility as once your request is accepted it will be in the trunc itself. But if for some reason it doesn't you're welcome to make PR for a trunc-compatible client I'm maintaining: https://bitbucket.org/citymania/cmclient

Reply:
As @ldpl says, when submitting a PR, if merged into master it will become an official feature, so no need to worry about that.

Reply:
> you're welcome to make PR for a trunk-compatible client I'm maintaining: https://bitbucket.org/citymania/cmclient

:0 it's the citymania client! I'd love to be included in it :)

Reply:
I guess you misunderstood my comment. This should be one DoCommandP(), just like CMD_CLEAR_AREA, just with an extra bit (set in p2) to say "only clear trees".

There should be no loops in the GUI code to send commands.

Reply:
Oh, can DoCommandP() be modified like that already? While keeping compatibility with the trunk server?

Reply:
Please reread what @ldpl and I wrote?

Reply:
I guess I'll try to explain it a bit more. There are two different ways of writing patches:
1) Make a patch for OpenTTD master/trunk. It should keep savegame compatibility but not network compatibility. You're never supposed to be able to connect with newer version of the game to older servers. So you should implement your patch in a most logical and clear way. E.g. modifying network commands instead of trying to avoid that. This is a type of patch that you can submit as PR for OpenTTD but you won't be able to use it on older servers.
2) Make compatible patch. It can't alter savegames at all and should keep network compatibility. Usually it means doing things in a hacky non-optimal way. Patch like that is not meant to be included into OpenTTD master so it makes no sense to submit it as a PR. But there are some patchpacks like CityMania client that specialize on patches like that so you can try making PR for them.

Sometimes there are patches that fit both categories but usually that only applies to something cosmetic that doesn't affect actual gameplay. 

In general you should aim for your patch to be included in OpenTTD and so follow route 1. Only if it's for some reason rejected you can try second option. Though there is usually a good chance that feature rejected by OpenTTD developers it's not a good feature in a first place so including it in patchpack is also pointless.

In this particular case your patch seems to follow route 2. And it's a fine feature in my opinion, so it can be included in CityMania client assuming code quality is ok (I haven't actually revieved it yet). But as I said you should try to make it included in master first. I generally don't include in CityMania client features that have a good chance to be accepted in master as it's basically a double work.